activejob/lib/active_job/arguments.rb

```diff
@@ -77,7 +77,7 @@ def serialize_argument(argument)
           result[WITH_INDIFFERENT_ACCESS_KEY] = serialize_argument(true)
           result
         when Hash
-          symbol_keys = argument.each_key.grep(Symbol).map(&:to_s)
+          symbol_keys = argument.each_key.grep(Symbol).map(&:to_s).sort
```

Comment:
This should not need to change to make a test pass. This is making production code slower. I think we should sort on the test side.